kafka-rest/src/main/java/io/confluent/kafkarest/ProducerMetrics.java

```diff
@@ -0,0 +1,390 @@
+
+    public SensorBuilder(ProduceMetricMBean bean, String jmxPrefix, String name) {
+      this.bean = bean;
+      this.sensor =
+          bean.sensor(String.join(":", jmxPrefix, ProducerMetricsRegistry.GROUP_NAME, name));
+    }
+
+    public SensorBuilder addAvg(String name, String doc) {
+      MetricName metricName = getMetricName(name, doc);
+      sensor.add(metricName, MeasuredStatSupplier.avg());
+      return this;
+    }
+
+    public SensorBuilder addRate(String name, String doc) {
+      MetricName metricName = getMetricName(name, doc);
+      sensor.add(metricName, MeasuredStatSupplier.rate());
+      return this;
+    }
+
+    public SensorBuilder addMax(String name, String doc) {
+      MetricName metricName = getMetricName(name, doc);
+      sensor.add(metricName, MeasuredStatSupplier.max());
+      return this;
+    }
+
+    public SensorBuilder addWindowedCount(String name, String doc) {
+      MetricName metricName = getMetricName(name, doc);
+      sensor.add(metricName, MeasuredStatSupplier.windowedCount());
+      return this;
+    }
+
+    public SensorBuilder addPercentiles(
+        String name, Map<String, Double> suffixPercentiles, String doc) {
+      Map<MetricName, Double> namePercentiles =
+          suffixPercentiles.entrySet().stream()
+              .map(
+                  suffixPercentile ->
+                      new AbstractMap.SimpleImmutableEntry<MetricName, Double>(
+                          getMetricName(name + suffixPercentile.getKey(), doc),
+                          suffixPercentile.getValue()))
+              .collect(Collectors.toMap(e -> e.getKey(), e -> e.getValue()));
+
+      sensor.add(MeasuredStatSupplier.percentiles(namePercentiles));
+      return this;
+    }
+
+    public Sensor build() {
+      return sensor;
+    }
+
+    private MetricName getMetricName(String name, String doc) {
+      MetricName metricName =
+          bean.metricName(
+              new MetricNameTemplate(
+                  name, ProducerMetricsRegistry.GROUP_NAME, doc, Collections.emptySet()));
+      return metricName;
+    }
+  }
+
+  public static class MeasuredStatSupplier {
+
+    public static MeasurableStat avg() {
+      return new Avg();
+    }
+
+    public static MeasurableStat rate() {
+      return new Rate();
+    }
+
+    public static MeasurableStat max() {
+      return new Max();
+    }
+
+    public static MeasurableStat windowedCount() {
+      return new WindowedCount();
+    }
+
+    public static CompoundStat percentiles(Map<MetricName, Double> percentiles) {
+      return new Percentiles(
+          30 * 1000 * 4,
+          30 * 1000,
+          Percentiles.BucketSizing.CONSTANT,
+          percentiles.entrySet().stream()
+              .map(
+                  namePercentile ->
+                      new Percentile(namePercentile.getKey(), namePercentile.getValue()))
+              .toArray(Percentile[]::new));
+    }
+  }
+
+  public static class MetricsBuilder {
```

Comment:
Do we need this class after the latest changes? Either I'm missing something, or it can be replaced by a simple `private` method now.

Reply:
iirc it blows up checkstyle with too many instantiations.

Reply:
That really seems odd to me. I would prefer not having a separate class for satisfying what seems to be a weird rule, but I don't think this should hold up the changes.